src/module-loader.js

```diff
@@ -92,12 +96,36 @@ export class ModuleLoader extends EventTarget {
         nodePath.join(this.basePath, directory, file.name)
       );
 
-      this.registry.add(
-        `/${nodePath.join(directory, nodePath.parse(file.name).name)}`,
-        endpoint
-      );
+      if (file.name.includes("$context")) {
+        this.addContext(directory, endpoint);
+      } else {
+        this.registry.add(
+          `/${nodePath.join(directory, nodePath.parse(file.name).name)}`,
+          endpoint
+        );
+      }
     });
 
     await Promise.all(imports);
   }
+
+  addContext(directory, endpoint) {
+    this.contextRegistry.add(`/${directory}`, endpoint.default);
+  }
```

Comment:
This function isn't necessary. It's only called in one place.

- [ ] inline the call to `this.contextRegistry.add()